pkg/apic/service.go

```diff
@@ -144,49 +162,51 @@ func (c *ServiceClient) postAPIServiceUpdate(serviceBody *ServiceBody) {
 
 // updateAPIServiceRevisionTitle - update title after creating or updating APIService Revision according to the APIServiceRevision Pattern
 func (c *ServiceClient) updateAPIServiceRevisionTitle(serviceBody *ServiceBody) string {
-	title := c.cfg.GetAPIServiceRevisionPattern() // "{{APIServiceName}} - {{date:YYYY/MM/DD}} - r {{revision}}"
-	revision := strconv.Itoa(serviceBody.serviceContext.revisionCount + 1)
+	apiSvcRevPattern := c.cfg.GetAPIServiceRevisionPattern() // "{{APIServiceName}} - {{date:YYYY/MM/DD}} - r {{revision}}"
+	dateRegEx := regexp.MustCompile(`{{date:.*?}}`)
 
-	replaceVars := map[string]string{"APIServiceName": serviceBody.APIName, "revision": revision}
-	// replace occurrences of `APIServiceName` and `revision` in title
-	for k, v := range replaceVars {
-		title = strings.Replace(title, fmt.Sprintf("{{%s}}", k), v, -1)
-	}
+	var dateFormat = ""
 
-	dateRegEx := regexp.MustCompile(`{{date:.*?}}`)
-	if dateRegEx.MatchString(title) {
-		var createdOn time.Time
-		if serviceBody.serviceContext.previousRevision == nil {
-			createdOn = time.Now()
-		} else {
-			createdOn = time.Time(serviceBody.serviceContext.previousRevision.ResourceMeta.Metadata.Audit.CreateTimestamp)
+	if dateRegEx.MatchString(apiSvcRevPattern) {
+		datePattern := dateRegEx.FindString(apiSvcRevPattern) //{{date:YYYY/MM/DD}} or one of the validate formats from apiSvcRevTitleDateMap
+		index := strings.Index(datePattern, ":")              // get index of ":" (colon)
+		date := datePattern[index+1 : index+11]               // sub out "{{date:" and "}}" to get the format of the date only
+		dateFormat = apiSvcRevTitleDateMap[date]              // make sure dateFormat is a valid date format
+		if dateFormat == "" {
+			log.Tracef("CENTRAL_APISERVICEREVISIONPATTERN is returning an invalid {{date:*}} format. Setting format to YYYY-MM-DD")
+			dateFormat = "2006/01/02"
 		}
-		year := strconv.Itoa(createdOn.Year())
-		month := createdOn.Format("01")
-		day := strconv.Itoa(createdOn.Day())
-
-		date := dateRegEx.FindString(title)
-		replaceDate := map[string]string{"MM": month, "DD": day, "YYYY": year}
-		for k, v := range replaceDate {
-			date = strings.Replace(date, k, v, -1)
-		}
-		date = strings.TrimPrefix(date, "{{date:")
-		date = strings.TrimSuffix(date, "}}")
+	}
 
-		//swap the date pattern with the date variable in the title
-		title = strings.Replace(title, dateRegEx.FindString(title), date, -1)
+	// create apiservicerevision template
+	apiSvcRevTitleTemplate := APIServiceRevisionTitle{
+		serviceBody.APIName,
+		time.Now().Format(dateFormat),
+		strconv.Itoa(serviceBody.serviceContext.revisionCount + 1),
 	}
-	return title
+
+	title, err := template.New("apiSvcRevTitle").Parse("{{.Name}} - {{.Date}} - r {{.Revision}}")
```

Comment:
Should this string be declared as a constant somewhere?

Reply:
Can be... will do.